.github/workflows/release.yml

```
@@ -68,14 +68,40 @@ jobs:
         env:
           SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT_STORE_TOKEN }}
 
+      - name: Import GPG key
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        id: import_gpg
+        run: |
+          GPG_KEY_FILE=/tmp/signing-key.gpg
+          echo "${{ secrets.GPG_PRIVATE_KEY }}" | base64 --decode > $GPG_KEY_FILE
+          
+          echo "${{ secrets.GPG_PRIVATE_KEY }}" | base64 --decode | gpg --import
+          FINGERPRINT=$(gpg --list-secret-keys --keyid-format LONG | awk '/^sec/{sub(/.*\//, "", $2); print $2; exit}')
+          
+          echo "::set-output name=fingerprint::${FINGERPRINT}"
+          echo "::set-output name=file::${GPG_KEY_FILE}"
+        env:
+          GPG_TTY: /dev/ttys000  # Set the GPG_TTY to avoid issues with pinentry
+
       - name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v4
         with:
           version: v1.21.2
           args: release --clean --timeout=60m --snapshot=${{ !startsWith(github.ref, 'refs/tags/v') }}
         env:
+          GPG_FINGERPRINT: ${{ steps.import_gpg.outputs.fingerprint }}
+          GPG_KEY_FILE: ${{steps.import_gpg.outputs.file }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT_STORE_TOKEN }}
+
+      - name: Remove GPG key
+        if: always()
```

```suggestion
      - name: Remove GPG key
        needs: [ import_gpg ]
        if: always()
```
The needs tag is missing on this job making `${{ steps.import_gpg.outputs.file }}` not working

---

@janosdebugs Nevermind, i'm stupid, `needs` is only for between jobs, not between steps......